### PR TITLE
Feature/cc mt service authentication

### PR DESCRIPTION
**Pull request by MSurvonen**

### CODE REVIEW

File: cc-mt-service/.env.example [hunk 1]
```diff
@@ -1 +1,9 @@
+<<<<<<< HEAD
```
Comment (bunjovis) [inline cc-mt-service/.env.example]: this file has merge conflicts, it should be resolved. Only one port and remove the other merge conflict related stuff (<<<<< HEAD, =======, >>>>>>> 043e03....)

File: cc-mt-service/src/app.ts [hunk 2]
```diff
@@ -1,21 +1,67 @@
...
+    var qs = require('qs');
+    var data = qs.stringify({
+      'client_secret': secret,
+      'client_id': client,
+      'grant_type': 'client_credentials',
+      'scope': scope
+    });
+    var config = {
+      method: 'post',
+      url: 'https://test-auth.materiaalitori.fi/connect/token?grant_type=client_credentials&scope=ym-tietoalusta-api',
```
Comment (bunjovis) [inline cc-mt-service/src/app.ts]: scope in the url should come from the variable scope.

**Review by bunjovis — CHANGES_REQUESTED** [review]

**Review by OpaJaa — DISMISSED** [review]
For me these look good.

File: cc-mt-service/src/app.ts [hunk 3]
```diff
@@ -1,21 +1,67 @@
...
+/**
+ * Get token to use test.materiaalitori.fi API to make possible to add adverts
+ * @param client 
+ * @param secret 
+ * @param scope 
+ * @returns token to Materiaalitori API
+ */
+async function getToken(client:string, secret:string, scope:string) {
+  try {
+    var axios = require('axios');
```
Comment (rvsape) [inline cc-mt-service/src/app.ts]: imports from packages should be on top of the file, plus using import syntax and const instead of var is preferred

Comment (rvsape) [reply cc-mt-service/src/app.ts]: npm run check gives some errors regarding this

File: cc-mt-service/src/app.ts [hunk 4]
```diff
@@ -1,21 +1,66 @@
...
+/**
+ * Get token to use test.materiaalitori.fi API to make possible to add adverts
+ * @param client 
+ * @param secret 
+ * @param scope 
+ * @returns token to Materiaalitori API
+ */
+async function getToken(client:string, secret:string, scope:string) {
+  try {
+    var data = qs.stringify({
```
Comment (rvsape) [inline cc-mt-service/src/app.ts]: const/let instead of var

**Review by rvsape — CHANGES_REQUESTED** [review]
- fix **npm run check** errors
- **npm run dev** error: 'ts-node' is not recognized as an internal or external command,

**Comment by MSurvonen** [comment]
@rvsape tested "npm run dev" and works for me normally. What to do because no errors to fix for me

**Comment by alihussain03** [comment]
For me it looks fine. i have tested using npm start and npm run check. npm run dev is giving error which needs to be fix 
image

**Comment by rvsape** [comment]
> 

missing dependency? npm i ts-node

**Review by rvsape — APPROVED** [review]

**Review by bunjovis — APPROVED** [review]

**Pull request merged**